test/unit/collector/api.test.js

```diff
@@ -74,6 +75,51 @@ tap.test('reportSettings', (t) => {
       t.end()
     })
   })
+
+  t.test('handles excessive payload sizes without blocking subsequent sends', (t) => {
```

Comment:
this implicitly tests the issue. not sure if it's good enough but here's what i'd do

```js
  t.test('handles excessive payload sizes without blocking subsequent sends', (t) => {
    const tstamp = 1_707_756_300_000 // 2024-02-12T11:45:00.000-05:00
    function log(data) {
      return JSON.stringify({
        level: 30,
        time: tstamp,
        pid: 17035,
        hostname: 'test-host',
        msg: data
      })
    }

    const kb512 = log(crypto.randomBytes(524_288).toString('base64'))
    const mb1 = log(crypto.randomBytes(1_048_576).toString('base64'))
    const toFind = log('find me')

    let apiHits = 0
    let sends = 0
    nock(URL)
      .persist(true)
      .post(helper.generateCollectorPath('log_event_data', RUN_ID))
      .reply(200, () => {
        apiHits += 1
      })

    agent.logs.add(kb512)
    agent.logs.send()
    agent.logs.add(mb1)
    agent.logs.send()
    agent.logs.add(toFind)
    agent.logs.send()

    agent.logs.on('finished log_event_data data send.', (t) => {
      sends++
      if (sends === 3) {
        t.equal(apiHits, 2, 'should only transmit batches that are within max limit')
      }
    })

  })
```

Reply:
you have to call send after every log enqueue because that's when data is harvested.  so this asserts once all 3 batches have been harvested that only 2 make it to the collector and implicitly doesn't back it up like it did before your fix

Reply:
i understand.  this is what our specifications state.  Unfortunately one bad apple will ruin the entire batch in this situation.